backend/server.js

```diff
@@ -23,6 +83,81 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+app.get('/thoughts', authenticateUser)
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find()
+  res.json({success: true, thoughts})
+})
+
+app.post('/thoughts', (req, res) => {
+  const { username, message } = req.body
+  
+  try {
+    const newThought = new Thought({
+      username, 
+      message
+    })
+    newThought.save()
+    res.status(201).json({success: true, id: newThought._id, username: newThought.username, createdAt: newThought.createdAt, message: newThought.message})
+  } catch (error) {
+    res.status(400).json({success: false, message: 'Could not post thought', error})
+  }
+  
+})
+
+app.post('/signup', async (req, res) => {
+  const { username, email, password } = req.body
+
+  try {  
+    const salt = bcrypt.genSaltSync()
+
+    const user = new User({
+      username, 
+      email, 
+      password: bcrypt.hashSync(password, salt)
+    }).save()
+    
+    res.status(201).json({success: true, id: user._id, username: user.username, email: user.email, accessToken: user.accessToken})
+    
+    
+    
+  } catch (error) {
+    res.status(400).json({success: false, message: 'Could not create user', error})
+  }
+})
+
+app.post('/sessions', async (req, res) => {
+  const { username, password } = req.body
+
+  try {
+    const user = await User.findOne({ username })
+
+    if (user && bcrypt.compareSync(password, user.password)) {
+      res.json({success: true, id: user._id, username: user.username, email: user.email,accessToken: user.accessToken})
+    } else {
+      res.json({success: false, notFound: true})
+    }
+  } catch (error) {
+    res.status(400).json({ success: false, message: 'Invalid request', error });
+  }
+})
+
+app.patch('/sessions/:id', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const updateUser = await User.findByIdAndUpdate(id, { fullName: req.body.fullName, age: req.body.age, location: req.body.location, description: req.body.description })
+
+    if (updateUser) {
+      res.json({success: true, updateUser})
+    } else {
```

Comment:
The user should be able to see the information added in the profile section? Or at least get a confirmation that it was saved. Now nothing happens when you press the 'add'-button.